File: java-tutorial/consumer/src/main/java/net/wissmueller/kafkatutorial/consumer/KafkaConsumer.java
package net.wissmueller.kafkatutorial.consumer;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class KafkaConsumer {
  private static final Logger log = LoggerFactory.getLogger(KafkaConsumer.class);

  @KafkaListener(topics = "timestamp")
  void listener(String timestamp) {
    log.info("Received: {}", timestamp);
  }
}

File: kotlin-tutorial/src/main/kotlin/com/example/kotlintutorial/KafkaConsumer.kt
package com.example.kotlintutorial

import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
class KafkaConsumer {

    @KafkaListener(topics = ["tutorial-topic"])
    fun processMessage(event: TimestampEvent) {
        println("Received: ${event.timestamp.toString()}")
    }

}